% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/eventCallers.R
\name{proximity}
\alias{proximity}
\usage{
proximity(
  gg,
  query,
  subject,
  ignore.strand = TRUE,
  verbose = F,
  mc.cores = 1,
  strict.ref = FALSE,
  chunksize = NULL,
  max.dist = 1e+06
)
}
\arguments{
\item{gg}{gGraph of the "alternate genome"}

\item{query}{GRanges of "intervals of interest" eg regulatory elements}

\item{subject}{GRanges of "intervals of interest" eg genes}

\item{ignore.strand}{whether to ignore strand of input GRanges. Default: True}

\item{verbose}{logical flag, verbose output. Default: False}

\item{mc.cores}{how many cores to use for the path exploration step or if
chunksize is provided, across chunks. Default: 1}

\item{strict.ref}{Boolean, only use actual reference edges in graph.
Default: False}

\item{chunksize}{chunks to split subject and query into to minimize memory
usage, if mc.cores>1 then each chunk will be allotted a core. Default: NULL}

\item{max.dist}{maximum genomic distance to store and compute (1MB by default)
should the maximum distance at which biological interactions may occur.
Default: 1e6}

\item{ref}{gGraph of the "reference genome", by is the reference genome but can be any gGraph}
}
\value{
gWalk object each representing a proximity
}
\description{
Takes a set of n "query" elements (GRanges object, e.g. genes) and determines
their proximity to m "subject" elements (GRanges object, e.g. regulatory
elements) subject to set of rearrangement adjacencies (GRangesList with width 1 range pairs)
}
\details{
This analysis makes the (pretty liberal) assumption that all pairs of adjacencies
that can be linked on a gGraph path are in cis (i.e. share a chromosome) in
the tumor genome.
}

Each output proximity is a gWalk that connects query-subject on the genome
described by gGraph gg.  Each gWalk is  annotated by the metadata of the
corresponding query-subject GRanges pair as well as fields "altdist" and "refdist"
specifying the "alternate and "reference" gGraph distance of the
query-subject pair.  The gWalk metadata field "reldist" specifies the relative
distance (i.e. ratio of altdist to refdist) for that walk.

For more details follow the Proximity Analysis in the gGnome Tutorial:

\href{http://mskilab.com/gGnome/tutorial.html#Proximity_analysis}{Proximity Analysis}
}
